# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1363199594 14400
#      Wed Mar 13 14:33:14 2013 -0400
# Node ID faaf60aa457ced89309c353eb4c21849f575c9a7
# Parent  00da706fba4807446d1b263017e39f5d14e33067
use separate automake convenience library for bison and flex generated files

* libinterp/Makefile.am (DIST_SRC): Include $(PARSER_SRC).
(liboctinterp_la_LIBADD): Include parse-tree/libparser.la.
* libinterp/parse-tree/module.mk (PARSE_TREE_INC): Remove
$(PARSER_INC) from the list.
(PARSE_TREE_SRC): Remove $(PARSER_SRC) from the list.
(noinst_LTLIBRARIES): Include parse-tree/libparser.la in the list.
(parse_tree_libparser_la_SOURCES, parse_tree_libparser_la_CPPFLAGS,
parse_tree_libparser_la_CXXFLAGS): New varibles.  Filter out
-Wold-style-cast from CXXFLAGS for parser convenience library.

diff --git a/libinterp/Makefile.am b/libinterp/Makefile.am
--- a/libinterp/Makefile.am
+++ b/libinterp/Makefile.am
@@ -120,16 +120,17 @@ nodist_octinclude_HEADERS = \
   interpfcn/graphics.h \
   oct-conf.h \
   version.h
 
 DIST_SRC = \
   octave.cc \
   $(OCTAVE_VALUE_SRC) \
   $(PARSE_TREE_SRC) \
+  $(PARSER_SRC) \
   $(INTERP_CORE_SRC) \
   $(INTERPFCN_SRC) \
   $(COREFCN_SRC)
 
 noinst_LTLIBRARIES =
 
 include parse-tree/module.mk
 include octave-value/module.mk
@@ -173,16 +174,17 @@ nodist_liboctinterp_la_SOURCES = \
 
 liboctinterp_la_CPPFLAGS = @OCTINTERP_DLL_DEFS@ $(AM_CPPFLAGS)
 
 include link-deps.mk
 
 liboctinterp_la_LIBADD = \
   octave-value/liboctave-value.la \
   parse-tree/libparse-tree.la \
+  parse-tree/libparser.la \
   interp-core/libinterp-core.la \
   interpfcn/libinterpfcn.la \
   corefcn/libcorefcn.la \
   $(top_builddir)/liboctave/liboctave.la \
   $(LIBOCTINTERP_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool manual:
 liboctinterp_current = 1
diff --git a/libinterp/parse-tree/module.mk b/libinterp/parse-tree/module.mk
--- a/libinterp/parse-tree/module.mk
+++ b/libinterp/parse-tree/module.mk
@@ -6,23 +6,16 @@ PARSER_INC = \
   parse-tree/lex.h \
   parse-tree/parse.h \
   parse-tree/parse-private.h
 
 PARSER_SRC = \
   parse-tree/lex.ll \
   parse-tree/oct-parse.yy
 
-## FIXME: Automake does not support per-object rules.
-##        These rules could be emulated by creating a new convenience
-##        library and using per-library rules.  Or we can just live
-##        with the extra warnings about old-sytle-casts. (09/18/2012)
-#lex.lo lex.o oct-parse.lo oct-parse.o: \
-#  AM_CXXFLAGS := $(filter-out -Wold-style-cast, $(AM_CXXFLAGS))
-
 PARSE_TREE_INC = \
   parse-tree/pt-all.h \
   parse-tree/pt-arg-list.h \
   parse-tree/pt-array-list.h \
   parse-tree/pt-assign.h \
   parse-tree/pt-binop.h \
   parse-tree/pt-bp.h \
   parse-tree/pt-cbinop.h \
@@ -43,18 +36,17 @@ PARSE_TREE_INC = \
   parse-tree/pt-mat.h \
   parse-tree/pt-misc.h \
   parse-tree/pt-pr-code.h \
   parse-tree/pt-select.h \
   parse-tree/pt-stmt.h \
   parse-tree/pt-unop.h \
   parse-tree/pt-walk.h \
   parse-tree/pt.h \
-  parse-tree/token.h \
-  $(PARSER_INC)
+  parse-tree/token.h
 
 PARSE_TREE_SRC = \
   parse-tree/pt-arg-list.cc \
   parse-tree/pt-array-list.cc \
   parse-tree/pt-assign.cc \
   parse-tree/pt-binop.cc \
   parse-tree/pt-bp.cc \
   parse-tree/pt-cbinop.cc \
@@ -74,18 +66,17 @@ PARSE_TREE_SRC = \
   parse-tree/pt-loop.cc \
   parse-tree/pt-mat.cc \
   parse-tree/pt-misc.cc \
   parse-tree/pt-pr-code.cc \
   parse-tree/pt-select.cc \
   parse-tree/pt-stmt.cc \
   parse-tree/pt-unop.cc \
   parse-tree/pt.cc \
-  parse-tree/token.cc \
-  $(PARSER_SRC)
+  parse-tree/token.cc
 
 ## Special rules for sources which must be built before rest of compilation.
 
 ## Don't use a pipeline to process gperf output since if gperf
 ## is missing but sed is not, the exit status of the pipeline
 ## will still be success and we will end up creating an empty
 ## oct-gperf.h file.
 parse-tree/oct-gperf.h: parse-tree/octave.gperf
@@ -101,12 +92,19 @@ parse-tree/oct-parse.yy: parse-tree/oct-
         esac; \
         case "$(BISON_PUSH_PULL_DECL_STYLE)" in \
           *dash*) decl="%define api.push-pull $${quote}both$${quote}"; ;; \
           *underscore*) decl="%define api.push_pull $${quote}both$${quote}"; ;; \
         esac; \
 	$(SED) "s/%PUSH_PULL_DECL%/$$decl/" $< > $@-t
 	mv $@-t $@
 
-noinst_LTLIBRARIES += parse-tree/libparse-tree.la
+noinst_LTLIBRARIES += \
+  parse-tree/libparse-tree.la \
+  parse-tree/libparser.la
 
 parse_tree_libparse_tree_la_SOURCES = $(PARSE_TREE_SRC)
 parse_tree_libparse_tree_la_CPPFLAGS = $(liboctinterp_la_CPPFLAGS)
+
+parse_tree_libparser_la_SOURCES = $(PARSER_SRC)
+parse_tree_libparser_la_CPPFLAGS = $(liboctinterp_la_CPPFLAGS)
+parse_tree_libparser_la_CXXFLAGS = \
+  $(filter-out -Wold-style-cast, $(AM_CXXFLAGS))
